build.rs

```diff
@@ -75,93 +114,88 @@ fn byte_array_to_os_string(bytes: &[u8]) -> OsString {
     OsString::from_wide(&wide)
 }
 
-fn probe_r_paths() -> io::Result<InstallationPaths> {
-    // First we locate the R home
-    let r_home = match env::var_os("R_HOME") {
-        // If the environment variable R_HOME is set we use it
-        Some(s) => PathBuf::from(s),
-
-        // Otherwise, we try to execute `R` to find `R_HOME`. Note that this is
-        // discouraged, see Section 1.6 of "Writing R Extensions"
-        // https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Writing-portable-packages
-        _ => {
-            let rout = Command::new("R")
-                .args(&[
-                    "-s",
-                    "-e",
-                    r#"cat(normalizePath(R.home()))"#
-                ])
-                .output()?;
-
-            let rout = byte_array_to_os_string(&rout.stdout);
-            if !rout.is_empty() {
-                PathBuf::from(rout)
-            } else {
-                return Err(Error::new(ErrorKind::Other, "Cannot find R home."));
-            }
-        }
-    };
+// Execute an R script and return the captured output
+fn r_command<S: AsRef<OsStr>>(r_binary: S, script: &str) -> io::Result<OsString> {
+    let out = Command::new(r_binary)
+        .args(&["-s", "-e", script])
+        .output()?;
+
+    // if there are any errors we print them out, helps with debugging
+    if !out.stderr.is_empty() {
+        println!(
+            "> {}",
+            byte_array_to_os_string(&out.stderr)
+                .as_os_str()
+                .to_string_lossy()
+        );
+    }
 
-    // Now the library location. On Windows, it depends on the target architecture
+    Ok(byte_array_to_os_string(&out.stdout))
+}
+
+// Get the path to the R home either from an envvar or by executing the actual R binary on PATH.
+fn get_r_home() -> io::Result<PathBuf> {
+    // If the environment variable R_HOME is set we use it
+    if let Some(r_home) = env::var_os(ENVVAR_R_HOME) {
+        return Ok(PathBuf::from(r_home));
+    }
+
+    // Otherwise, we try to execute `R` to find `R_HOME`. Note that this is
+    // discouraged, see Section 1.6 of "Writing R Extensions"
+    // https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Writing-portable-packages
+    let rout = r_command("R", r#"cat(normalizePath(R.home()))"#)?;
+    if !rout.is_empty() {
+        Ok(PathBuf::from(rout))
+    } else {
+        Err(Error::new(ErrorKind::Other, "Cannot find R home."))
+    }
+}
```

Comment:
This fragment is duplicated at least twice, perhaps you can move `rout` check into `r_command()`? Or make another function that calls `r_command()` and processes output?

Reply:
As the error messages are a bit different, I prefer to leave this.